editor/gui/editor_spin_slider.cpp

```diff
@@ -221,30 +220,23 @@ void EditorSpinSlider::_value_input_gui_input(const Ref<InputEvent> &p_event) {
 		}
 
 		Key code = k->get_keycode();
+
 		switch (code) {
-			case Key::UP: {
+			case Key::UP:
+			case Key::DOWN:
+			{
 				_evaluate_input_text();
 
 				double last_value = get_value();
-				set_value(last_value + step);
-				double new_value = get_value();
-
-				if (new_value < CLAMP(last_value + step, get_min(), get_max())) {
-					set_value(last_value + real_step);
+				if (code == Key::DOWN) {
+					step *= -1;
 				}
-
-				value_input_dirty = true;
-				set_process_internal(true);
-			} break;
-			case Key::DOWN: {
-				_evaluate_input_text();
-
-				double last_value = get_value();
-				set_value(last_value - step);
+				set_value(last_value + step);
 				double new_value = get_value();
-
-				if (new_value > CLAMP(last_value - step, get_min(), get_max())) {
-					set_value(last_value - real_step);
+				
+				double clamp_value = CLAMP(new_value, get_min(), get_max());
+				if (new_value < clamp_value || new_value > clamp_value) {
```

Comment:
This is just `new_value != clamp_value` isn't it?

Reply:
yes that's better, I checked, `set_value` is only executed when limits are reached.